okhttp/src/main/java/okhttp3/internal/connection/RealCall.kt

```diff
@@ -235,17 +239,15 @@ class RealCall(
   }
 
   /** Finds a new or pooled connection to carry a forthcoming request and response. */
-  internal fun prepareNewExchange(chain: RealInterceptorChain): Exchange {
+  internal fun initExchange(chain: RealInterceptorChain): Exchange {
     synchronized(connectionPool) {
       check(!noMoreExchanges) { "released" }
-      check(exchange == null) {
-        "cannot make a new request because the previous response is still open: " +
-            "please call response.close()"
-      }
+      check(exchange == null)
```

Comment:
why remove the explaining message?  Are we sure this will be caught in previous location?

Reply:
Yeah. By promoting the other check to occur unconditionally this one becomes more of an assert.